ansible/roles/master/tasks/main.yml

```diff
@@ -43,53 +37,26 @@
     controller_manager_token: "{{ tokens.results[0].content|b64decode }}"
     scheduler_token: "{{ tokens.results[1].content|b64decode }}"
     kubectl_token: "{{ tokens.results[2].content|b64decode }}"
-
-- name: write the config file for the controller-manager
-  template: src=controller-manager.j2 dest={{ kube_config_dir }}/controller-manager
-  notify:
-    - restart controller-manager
+    kubelet_token: "{{ tokens.results[3].content|b64decode }}"
 
 - name: write the kubecfg (auth) file for controller-manager
   template: src=controller-manager.kubeconfig.j2 dest={{ kube_config_dir }}/controller-manager.kubeconfig
   notify:
     - restart controller-manager
 
-- name: Enable controller-manager
-  service: name=kube-controller-manager enabled=yes state=started
-
-- name: write the config file for the scheduler
-  template: src=scheduler.j2 dest={{ kube_config_dir }}/scheduler
-  notify:
-    - restart scheduler
-
 - name: write the kubecfg (auth) file for scheduler
   template: src=scheduler.kubeconfig.j2 dest={{ kube_config_dir }}/scheduler.kubeconfig
   notify:
     - restart scheduler
 
-- name: Enable scheduler
-  service: name=kube-scheduler enabled=yes state=started
-
 - name: write the kubecfg (auth) file for kubectl
   template: src=kubectl.kubeconfig.j2 dest={{ kube_config_dir }}/kubectl.kubeconfig
 
-# Enable kubelet on master only when OpenContrail is in use; see
-# https://github.com/kubernetes/contrib/pull/183
-- name: write the config files for kubelet
-  template: src=kubelet.j2 dest={{ kube_config_dir }}/kubelet
-  notify:
-    - restart kubelet
-  when: networking == 'opencontrail'
-
-- name: Enable kubelet
-  service: name=kubelet enabled=yes state=started
-  when: networking == 'opencontrail'
-
-- name: write the delay-master-services target
-  copy: src=delay-master-services.target dest=/etc/systemd/system/ mode=0644
+- include: ha_master.yml
+  when: groups['masters']|length > 1
```

Comment:
If master is < 1, how will the the master services start with everything that was removed?